jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/AbstractHTTP2ServerConnectionFactory.java

```diff
@@ -245,12 +289,16 @@ public HttpConfiguration getHttpConfiguration()
     protected Map<Integer, Integer> newSettings()
     {
         Map<Integer, Integer> settings = new HashMap<>();
-        settings.put(SettingsFrame.HEADER_TABLE_SIZE, getMaxDynamicTableSize());
-        settings.put(SettingsFrame.INITIAL_WINDOW_SIZE, getInitialStreamRecvWindow());
-        int maxConcurrentStreams = getMaxConcurrentStreams();
-        if (maxConcurrentStreams >= 0)
-            settings.put(SettingsFrame.MAX_CONCURRENT_STREAMS, maxConcurrentStreams);
-        settings.put(SettingsFrame.MAX_HEADER_LIST_SIZE, getHttpConfiguration().getRequestHeaderSize());
+        int maxTableSize = getMaxDecoderTableSize();
+        if (maxTableSize != 4096)
```

Comment:
Can we add a constant for this 4096 value.

Reply:
Is it possible to combine this logic with the `settings.compute` code you have elsewhere?

Reply:
Not sure what logic you refer to, but client and server have different logics and pull the values from different places (the `HTTP2Client` and the `AbstractHTTP2ServerConnectionFactory`), so would be difficult to share the code.

Reply:
Now using a constant for the default maxTableSize of 4096.